### PR TITLE
ST: add test for maintenance window

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- New Test

### Description

This PR introduce new system test, which check correct function of [Maintenance window](https://strimzi.io/docs/master/#assembly-maintenance-time-windows-deployment-configuration-kafka). This test doesn't check manual trigger of rolling update, only automatic trigger for CA renewal.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests testcase=SecurityST

File: systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java [hunk 1]
```diff
@@ -422,6 +425,66 @@ private String createSecret(String resourceName, String secretName, String keyNa
...
+        assertThat("Rolling update was performed out of maintenance window!", kafkaPods, is(StUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME))));
+
+        LOGGER.info("Wait until rolling update is triggered during maintenance window");
+        StUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        assertThat("Rolling update wasn't performed in correct time", compareMaintenanceTime(maintenanceStart, maintenanceStopHour));
+        waitForClusterAvailability(NAMESPACE);
+    }
+
+    private boolean compareMaintenanceTime(long minute, long hour) {
```
Comment (serrss) [inline systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: Tests, where data is used, are always very complex. And compare data as _long_, can confuse person why will update this test in the future. I guess the better way is having the variables as _LocalDateTime_ in this ST. And in that case, you are able to use _compareTo_ method for better time calculation. Java doc - https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#compareTo-java.time.chrono.ChronoLocalDateTime-

**Comment by Frawless** [comment]
@scholzj after several executions, I didn't hit any race condition in this 1 sec window, but it might be useful to complete this wait few seconds earlier just to be sure (before maintenance window starts)

File: systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java [hunk 2]
```diff
@@ -422,6 +425,56 @@ private String createSecret(String resourceName, String secretName, String keyNa
...
+                .withTls(false)
+                .endKafkaListenerExternalNodePort()
+                .endListeners()
+                .endKafka()
+                .addNewMaintenanceTimeWindow(maintenanceWindowCron)
+                .endSpec().done();
+
+        Map<String, String> kafkaPods = StUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Annotate secret {} with secret force renew annotation", secretName);
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java]: "... with certificate force-renew annotation"

**Pull request merged**